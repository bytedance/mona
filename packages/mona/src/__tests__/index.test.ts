import { createProjectConfig, createAppConfig, createPageConfig } from '../index';

describe('mona createConfig', () => {
  test('createProjectConfig', () => {
    expect(() =>
      createProjectConfig({
        projectName: 'mona',
        input: './src/app.tsx',
        output: 'dist',
      }),
    ).not.toThrow();
    expect(
      createProjectConfig({
        projectName: 'mona',
        input: './src/app.tsx',
        output: 'dist',
      }),
    ).toMatchSnapshot();
  });
  test('createProjectConfig', () => {
    expect(() =>
      createAppConfig({
        pages: [],
      }),
    ).not.toThrow();
    expect(
      createAppConfig({
        pages: [],
      }),
    ).toMatchSnapshot();
  });
  test('createProjectConfig', () => {
    expect(() => createPageConfig({})).not.toThrow();
    expect(createPageConfig({})).toMatchSnapshot();
  });
});
