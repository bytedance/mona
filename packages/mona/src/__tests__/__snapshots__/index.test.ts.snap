// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mona createConfig createProjectConfig 1`] = `
Object {
  "input": "./src/app.tsx",
  "output": "dist",
  "projectName": "mona",
}
`;

exports[`mona createConfig createProjectConfig 2`] = `
Object {
  "pages": Array [],
}
`;

exports[`mona createConfig createProjectConfig 3`] = `Object {}`;
